feat: use micro-stacks

aulneau opened the pull request:
Hi there 👋 

This was mostly an exploration for myself to see how hard it is to migrate an app from using `@stacks/*` to `micro-stacks/*` related libraries. Only took about a half hour to move things over!

Bundle size difference:

with micro-stacks

```sh
File sizes after gzip:

  250.98 kB  build/static/js/main.b1786ae0.js
  1.79 kB    build/static/js/787.e4ec841a.chunk.js
  227 B      build/static/css/main.eaa4c575.css
```

with stacks.js 

```sh
File sizes after gzip:

  483.17 kB (+232.2 kB)  build/static/js/main.7121c311.js
  305.88 kB              build/static/js/124.6c09110b.chunk.js
  1.79 kB                build/static/js/787.e4ec841a.chunk.js
  227 B                  build/static/css/main.eaa4c575.css
```

The advantage of using `micro-stacks` is a much lighter bundle size (10x+ smaller -- with exact API matches with core `@stacks/*` libs.) `micro-stacks` core only has two dependencies, and each of those have zero. Additionally, with `@micro-stacks/react` many of the things you were doing manually are handled by the library: current network state, account based on network, loading states, etc. 

I've tested most of this, but there might be some broken features :) happy to chat through any of this, and absolutely no pressure on switching over if that is something that is not interesting to you!

talhasch commented:
Hi @aulneau 

Thanks for the commit! One thing i see is some tests fail. Component snapshot tests depends on stacks/connect-react's Connect provider. You can see in helper/test-helper.tsx. Is it possible to replace it with micro-stacks?

I'll check all functionality of the app to make sure if all ok.

aulneau commented:
@talhasch I updated the dependencies in the tests, but I'm not able to run tests on `main` or this branch on my machine, i don't think the dependencies you have work with my m1 mac.

One other change I'd _highly_ recommend is moving away from create-react-app (react-scripts) and move to a more modern bundling framework like [vite](https://vitejs.dev/)

aulneau commented:
@talhasch I have a branch here https://github.com/fungible-systems/multisafe-ui/pull/1 using vite. you'll get many more benefits from using vite: faster builds, more modern outputs, MUCH better HMR, and much much more

talhasch commented:
@aulneau Thanks for letting know, I'll definitely check vite! 

Could you click on "allow edits from maintainers" checkbox. I have some code to push this repo to fix test problems.

aulneau commented:
> @aulneau Thanks for letting know, I'll definitely check vite!
> 
> Could you click on "allow edits from maintainers" checkbox. I have some code to push this PR to fix test problems.

I'm never able to find that setting 🤔 but I invited you as collaborator

talhasch commented on src/hooks/use-user-session.ts:
```diff
@@ -1,37 +1,9 @@
-    return [userSession, userData, openAuth, signOut];
+// just for migration
+const useUserSession = (): [null, boolean, () => void, () => void] => {
+    const {openAuthRequest, isSignedIn, signOut} = useAuth()
```
@aulneau can we keep UserSession and UserData here? We'll need them in the near future for gaia storage integration.

aulneau replied:
@talhasch we don't need userSession or userData for gaia with micro-stacks, there are hooks for that (`usePutFile`, `useGetFile`, and soon some more specific `useStorage` helpers).

Can you tell me more about how you want to use gaia? I'm working on some new tooling right now and having some real use cases in mind would be super helpful

talhasch replied:
This is early integration of gaia we disabled during development since it was slowing down progress: https://github.com/Trust-Machines/multisafe-ui/blob/main/src/hooks/use-storage.ts#L25

talhasch commented on src/views/create/index.tsx:
```diff
@@ -112,7 +113,7 @@ const Create = (_: RouteComponentProps) => {
                                 }}
                                 onNext={() => {
-                                    if (!userData) {
+                                    if (!isSignedIn) {
```
Once we nave the userData again, we can use it here as it was.

aulneau replied:
I'd suggest we keep it as `isSignedIn`, this is functionally the same, [see here](https://github.com/fungible-systems/micro-stacks/blob/main/packages/react/src/hooks/use-auth.ts#L36)

talhasch commented on src/views/create/index.tsx:
```diff
@@ -124,8 +125,7 @@ const Create = (_: RouteComponentProps) => {
 
-                                    doContractDeploy({
-                                        network: stacksNetwork,
+                                    openContractDeploy({
```
micro-stacks doesn't require network argument since it handles network updates itself i guess? right?

aulneau replied:
yes, that is correct!

talhasch commented on tsconfig.json:
```diff
@@ -1,24 +1,23 @@
-      "dom",
-      "dom.iterable",
-      "esnext"
+      "DOM",
```
@aulneau updates on tsconfig for micro-stacks libraries or improvements?

aulneau replied:
this was something I was testing when I had issues running the app -- it should make no difference, it's the default tsconfig from the react-ts vite starter (which we should use if you want to work with my other PR) :)

talhasch commented:
> > @aulneau Thanks for letting know, I'll definitely check vite!
> > Could you click on "allow edits from maintainers" checkbox. I have some code to push this PR to fix test problems.
> 
> I'm never able to find that setting 🤔 but I invited you as collaborator

Awesome! I pushed some updates. It should fix test issues. Could you try running tests? Also dropped some comments. After resolving them ready to merge 🎉

aulneau commented:
> > > @aulneau Thanks for letting know, I'll definitely check vite!
> > > Could you click on "allow edits from maintainers" checkbox. I have some code to push this PR to fix test problems.
> > 
> > 
> > I'm never able to find that setting 🤔 but I invited you as collaborator
> 
> Awesome! I pushed some updates. It should fix test issues. Could you try running tests? Also dropped some comments. After resolving them ready to merge 🎉

wonderful! I've ran the tests and they work now on my machine :)

I've also made a few new commits:

- https://github.com/Trust-Machines/multisafe-ui/pull/1/commits/567654c97a351e26ddd7b3907c67ce55ce3132ac: this just updates the deps for the app to newest, non-breaking versions
- https://github.com/Trust-Machines/multisafe-ui/pull/1/commits/f1770c8c8e349864f1ca0ec8b8124b03e56fe949: this is an example of how you can start to show more feedback to users on different actions, eg "Confirm in wallet" during the create a safe step
- https://github.com/Trust-Machines/multisafe-ui/pull/1/commits/dbba769be21885c791225f3546a28481df847cba: one difference between stacks.js connect and micro-stacks is you have to handle the "no wallet found" screen yourself. I added a simple html native dialog, but you should update this to use your dialog/modal components :)
